app/lib/actions/katello/product/destroy.rb

```diff
@@ -62,6 +62,10 @@ def humanized_name
           _("Delete Product")
         end
 
+        def clear_pool_associations
+          self.pool_products.delete_all
```

Comment:
This should be looking at a product and not self :)

Reply:
uhhhh  how did the test pass??? I will investigate that :)